### ISSUE TITLE
When will it be possible to use it in Online Mode?

### ISSUE THREAD

**Issue by Zero3K** [DoaJCBlogger/Talk32#4]
*subject*

**Comment by Orlean54** [comment]
same question

**Comment by Zero3K** [comment]
I think this project has been abandoned.

**Comment by Zero3K** [comment]
I see that a commit has been done to it recently. I hope that there are more commits soon that fix this and the other issues.

**Comment by Zero3K** [comment]
Try using https://github.com/BridgeTrollPW/openSonux.

**Comment by Zero3K** [comment]
1. I don't know. It just looks like its smaller in size than Libcurl.
2. Maybe https://github.com/zaphoyd/websocketpp will work.

**Comment by Zero3K** [comment]
https://stackoverflow.com/questions/2629421/how-to-use-boost-in-visual-studio-2010

**Comment by DoaJCBlogger** [comment]
Thanks for the StackOverflow link. I'll try that within the next few days.

I visited the WebSocket++ website and in the requirements it says "C++03 or later" so I should be able to add it to the program and enable online mode soon. Thank you for bringing that to my attention.

The program will probably not get much bigger in terms of file size or memory usage. Regarding external dependencies, the website mentions Asio, OpenSSL, and zlib. I already link with OpenSSL and zlib so it looks like I just have to compile and link Boost.

**Comment by DoaJCBlogger** [comment]
"How hard is it to enable Online Mode?" Probably not too hard once I add Boost and WebSocket++.

"Have you tried implementing it yet or is it still too soon to try?" It's definitely not too soon to try. My goal this weekend is to compile and link Boost and then get WebSocket++ working.

**Comment by DoaJCBlogger** [comment]
This is the code I added to main.cpp:
```
#include <websocketpp/config/boost_config.hpp>
#include <websocketpp/client.hpp>
typedef websocketpp::client<websocketpp::config::boost_client> client;
```
This is the error I'm getting:
```
main.cpp(24) : error C3083: 'config': the symbol to the left of a '::' must be a type
main.cpp(24) : error C2039: 'boost_client' : is not a member of 'websocketpp'
main.cpp(24) : error C2065: 'boost_client' : undeclared identifier
```

**Comment by Zero3K** [comment]
https://stackoverflow.com/questions/16386359/error-this-thread-the-symbol-to-the-left-of-a-must-be-a-type/21848611

https://www.codeproject.com/Questions/1178046/Solution-for-Cplusplus-error-c-is-not-a-member-of

http://www.cpp.re/forum/beginner/213242/

**Comment by pachuco** [comment]
Maybe borrow network code from Miranda NG.

**Comment by pachuco** [comment]
Because Miranda NG has a Discord plugin.

**Comment by Zero3K** [comment]
> The only thing I can think of right now is that there's a WebSocket++ header file that I didn't include.

That makes sense.

**Comment by NailYggdrago** [comment]
Still having the same issue or are you now having a different one?

**Comment by Zero3K** [comment]
Searching Google doesn't help? Also, did you try asking via an Issue on WebSocket++'s Repository?

**Comment by Zero3K** [comment]
Do you have the location of Boost as an entry in the list of paths it uses when including source code?

**Comment by DoaJCBlogger** [comment]
Yes. When I first added Boost and WebSocket++, I was getting errors because WebSocket++ included Boost headers using <> instead of "", which meant that I had to add Boost to my list of include directories. I knew it wouldn't be practical to replace all instances of "#include <boost/[header.hpp]>" with relative paths in quotes so I added the following line to my makefile:

`INCLUDE = $(INCLUDE);.\boost;.\websocketpp`

**Comment by DoaJCBlogger** [comment]
You mean using an older version of Boost? I'll try that.

**Comment by Zero3K** [comment]
Yes. It might work for you.

**Comment by DoaJCBlogger** [comment]
I built Boost 1.70.0 and the project is a lot closer to working. Now the linker says it can't find these methods:

- boost::throw_exception
- boost::asio::async_connect
- boost::asio::async_read_until

I'm going to see if I can build or find the libraries needed to resolve this.

**Comment by Zero3K** [comment]
https://stackoverflow.com/questions/9272648/boost-symbol-not-found
https://stackoverflow.com/questions/11718597/asioasync-connect-and-io-service-run-dont-compile
https://stackoverflow.com/questions/3058310/boostasioasync-read-until-problem
https://www.boost.org/doc/libs/1_77_0/doc/html/boost_asio/using.html

**Comment by Zero3K** [comment]
Did those links help any?

**Comment by edo9300** [comment]
Found this, and it looks like a nice POC, may i ask why you're using visual c++ 2010? up to visual studio 2017, there's the option to install the xp tolset, that targets up to windows xp service pack 3 (the toolset can be installed on any successive visual studio version as well). That would help solving problems with newer libraies that might require a newer c++ version, as you'll be able to go up to c++17. And if you're ok with using some workarounds, you can even make the program work up to windows 2000 while using the xp toolset. (The main project i work on, works out of the box on windows 2000 without any service packs and no extra dependencies, and is compiled using visual studio 2017.)

**Comment by DoaJCBlogger** [comment]
@edo9300: I use the Windows SDK 7.1 and VC++ 2010 because I want people to be able to build the project on Windows XP, and without having to install too much.

**Comment by edo9300** [comment]
@DoaJCBlogger makes sense 😀

**Comment by Zero3K** [comment]
@DoaJCBlogger What about my question?

**Comment by DoaJCBlogger** [comment]
I tried the solutions with BOOST_NO_EXCEPTIONS and defining throw_exception but it still doesn't work. I saw the other solutions that change how ASIO methods are called but I haven't tried them yet.

**Comment by Zero3K** [comment]
Okay.

**Comment by Zero3K** [comment]
Any news?

**Comment by DoaJCBlogger** [comment]
Not yet.

**Comment by Zero3K** [comment]
Maybe you could ask for help via Upwork.com (or on Stackoverflow.com).

**Comment by Zero3K** [comment]
Well?

**Comment by DoaJCBlogger** [comment]
I've been doing some other projects lately but I can ask about it on Discord.

**Comment by Zero3K** [comment]
Okay.

**Comment by DoaJCBlogger** [comment]
I just asked about this in 2 large Discord servers.

**Comment by Zero3K** [comment]
Could https://github.com/vldr/WskHttp work?

**Comment by DoaJCBlogger** [comment]
I don't think so.

**Comment by Zero3K** [comment]
I was able to compile it using Visual Studio 2019.

**Comment by DoaJCBlogger** [comment]
@Zero3K: Can you try building this version in VS 2019? It's what I sent the other programmer. https://www.dropbox.com/s/vuino7p86st7i6q/Talk32.7z?dl=0

**Comment by Zero3K** [comment]
[talk32-vs2019-errors.txt](https://github.com/DoaJCBlogger/Talk32/files/9485593/talk32-vs2019-errors.txt)

**Comment by DoaJCBlogger** [comment]
Thank you. Those errors look like what I get with VS 2015.

**Comment by Zero3K** [comment]
Any news?

**Comment by DoaJCBlogger** [comment]
Not yet.

**Comment by DoaJCBlogger** [comment]
Last night I was able to add the Discord API "heartbeat" and it was able to stay connected.

**Comment by DoaJCBlogger** [comment]
Tonight I was able to receive and display an incoming Discord message with Talk32. I'll commit and push the changes this week.
![talk32_first_received_message_on_windows_xp](https://user-images.githubusercontent.com/34990159/211234548-ef1c042c-cd58-4843-acd0-702c5e039c63.png)

**Comment by DoaJCBlogger** [comment]
It currently works pretty well in online mode so I'm going to close the issue
